src/applications/personalization/dashboard-2/components/apply-for-benefits/ApplyForBenefits.jsx

```diff
@@ -53,17 +73,150 @@ const ApplyForBenefits = ({ verifiedSavedForms }) => {
       {!verifiedSavedForms.length && (
         <p>You have no applications in progress.</p>
       )}
-    </div>
+    </>
+  );
+};
+
+const BenefitsOfInterest = ({ children, showChildren }) => {
+  return (
+    <>
+      <h3 className="vads-u-font-size--h4 vads-u-font-family--sans vads-u-margin-bottom--2p5">
+        Benefits you might be interested in
+      </h3>
+      <div
+        className="vads-l-grid-container vads-u-padding--0"
+        data-testid="benefits-of-interest"
+      >
+        {!showChildren && (
+          <div className="vads-u-margin-y--2">
+            <LoadingIndicator message="Loading benefits you might be interested in..." />
+          </div>
+        )}
+        {showChildren && <div className="vads-l-row">{children}</div>}
+        <AdditionalInfo triggerText="What benefits does the VA offer?">
+          <p>Explore VA.gov to learn about the benefits we offer</p>
+          <ul>
+            <li>
+              <a>blah</a>
+            </li>
+            <li>
+              <a>blahhhh</a>
+            </li>
+          </ul>
+        </AdditionalInfo>
+      </div>
+    </>
+  );
+};
+
+const ApplyForBenefits = ({
+  getDD4EDUStatus,
+  getESREnrollmentStatus,
+  hasDD4EDU,
+  hasLoadedAllData,
+  isInESR,
+  isPatient,
+  shouldGetDD4EDUStatus,
+  shouldGetESRStatus,
+  verifiedSavedForms,
+}) => {
+  useEffect(
+    () => {
+      if (shouldGetESRStatus) {
+        getESREnrollmentStatus();
+      }
+    },
+    [shouldGetESRStatus, getESREnrollmentStatus],
+  );
+
+  useEffect(
+    () => {
+      if (shouldGetDD4EDUStatus) {
+        getDD4EDUStatus();
+      }
+    },
+    [shouldGetDD4EDUStatus, getDD4EDUStatus],
+  );
+
+  const showHealthCare = !isPatient && !isInESR;
+  const showEducation = !hasDD4EDU;
+
+  return (
+    <>
+      <h2>Apply for benefits</h2>
+      <ApplicationsInProgress verifiedSavedForms={verifiedSavedForms} />
+      <BenefitsOfInterest showChildren={hasLoadedAllData}>
```

Comment:
...we could pass the `showHealthCare` and `showEducation` data to `BenefitsOfInterest` as props. But with this approach it:
- prevents one level of prop passing
- is maybe more readable and understandable because we can see more of what `BenefitsOfInterest` is going to render _without_ looking at its implementation

I got the idea from https://epicreact.dev/one-react-mistake-thats-slowing-you-down

Reply:
I like this approach 👍